### PR TITLE
Enables text send and read operations for WebSocket connections. 

### LINKED ISSUE #80: Can't write or read text data using the rendezvous connection
There isn't seem a way to write text data to the rendezvous connection.

```listener.acceptConnectionAsync().join()``` returns  ```HybridConnectionChannel```, which I can cast to ```WebSocketChannel``` as that is the underlying type. However, the method I need,  ```CompletableFuture<Void> writeAsync(Object data, Duration timeout, boolean isEnd, WriteMode mode)```, is not public. 

Similarly, the ``readTextAsync``` method is not public, hence there is no way to read text messages.

### PR DESCRIPTION
**Pull request by giventocode**
Closes #80

### CODE REVIEW

File: src/main/java/com/microsoft/azure/relay/ClientWebSocket.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ boolean isOpen() {
 	 * 
 	 * @return Returns a CompletableFuture which completes when websocket receives text messages
 	 */
-	CompletableFuture<String> readTextAsync() {
+	public CompletableFuture<String> readTextAsync() {
```
Comment (bainian12345) [inline src/main/java/com/microsoft/azure/relay/ClientWebSocket.java:159]: >public C [](http://example.com/codeflow?start=1&length=8)

Why do you need to make the two methods in this class public? `WebSocketChannel` is more public facing and `ClientWebSocket` is more of an internal implementation class, so I would prefer if you leave these methods internal.

Comment (bainian12345) [reply src/main/java/com/microsoft/azure/relay/ClientWebSocket.java:159]: You can expose public methods in `WebSocketChannel` class instead.

File: src/test/java/com/microsoft/azure/relay/SendReceiveTest.java [hunk 2]
```diff
@@ -73,6 +74,16 @@ public void websocketSmallSendSmallResponseTest() {
...
+	public void websocketSendTextResponseTextTest() {
+		CompletableFuture<Void> listenerTask = sendAndReceiveTextWithWebsocketListener(WS_MSG, WS_MSG);
+		CompletableFuture<Void> clientTask = sendAndReceiveTextWithWebsocketClient(WS_MSG, WS_MSG);
+				
+		listenerTask.join();
+		clientTask.join();
+	}
+
+
 	@Test
```
Comment (bainian12345) [inline src/test/java/com/microsoft/azure/relay/SendReceiveTest.java:87]: nit: remove the extra empty line

File: src/test/java/com/microsoft/azure/relay/SendReceiveTest.java [hunk 3]
```diff
@@ -73,6 +74,16 @@ public void websocketSmallSendSmallResponseTest() {
 		clientTask.join();
 	}
 
+	@Test
+	public void websocketSendTextResponseTextTest() {
+		CompletableFuture<Void> listenerTask = sendAndReceiveTextWithWebsocketListener(WS_MSG, WS_MSG);
```
Comment (bainian12345) [inline src/test/java/com/microsoft/azure/relay/SendReceiveTest.java:79]: I would like to see larger data (>64kb) be tested as well. Maybe you can create small and large strings out of the `SMALL_BYTES` and `LARGE_BYTES` from above.

File: src/main/java/com/microsoft/azure/relay/WebSocketChannel.java [hunk 4]
```diff
@@ -113,7 +113,17 @@ public void close() throws IOException {
...
-	
+	/**
+	 * Sends the text data to the remote endpoint within a timeout in one of the WriteModes.
+	 *
+	 * @param textData Text message to be sent.
+	 * @param timeout The timeout to connect to send the data within. May be null to indicate no timeout limit.
+	 * @return A CompletableFuture which completes when websocket finishes sending the data.
+	 * @throws TimeoutException Throws when the sending task does not complete within the given timeout.
+	 */
+	public CompletableFuture<Void> writeTextAsync(String textData, Duration timeout) {
```
Comment (bainian12345) [inline src/main/java/com/microsoft/azure/relay/WebSocketChannel.java:124]: Just wondering, is there any functional difference where you would need to use WriteMode.TEXT instead?

**Review by bainian12345 — CHANGES_REQUESTED** [review]
:clock1:

File: src/test/java/com/microsoft/azure/relay/SendReceiveTest.java (hunk 3 above)
Comment (giventocode) [reply src/test/java/com/microsoft/azure/relay/SendReceiveTest.java]: >64kb is not possible in a single message. The `DefaultMaxTextMessageBufferSize` is by default `RelayConstants.DEFAULT_CONNECTION_BUFFER_SIZE`, which is 64Kb. I added a test with a string of 64KB.

File: src/main/java/com/microsoft/azure/relay/WebSocketChannel.java (hunk 4 above)
Comment (giventocode) [reply src/main/java/com/microsoft/azure/relay/WebSocketChannel.java:124]: Perhaps if you want to control the encoding of the String. However, in our case there is no value in exposing the WriteMode or multi-part messages.

**Review by bainian12345 — APPROVED** [review]
:shipit:

**Pull request merged**